src/index.tsx

```diff
@@ -1,38 +1,51 @@
-
 import * as React from 'react'
 import * as ReactDOM from 'react-dom'
 import { Provider } from 'react-redux'
-import createHistory from 'history/createBrowserHistory'
-import { Route } from 'react-router'
+import { Route, Switch } from 'react-router'
 import { ConnectedRouter } from 'react-router-redux'
 import { Container } from 'semantic-ui-react'
 
 /*Pages Components Imports */
-import App from './App'
-import { LoginPage } from './routes/LoginPage'
-import Board from './components/Board'
 import Layout from './components/Layout'
 
+import { AUTH } from './services/auth'
+
 import registerServiceWorker from './registerServiceWorker'
 import './index.css'
 
-import store from './redux/store'
-export const history = createHistory()
+import store, { history } from './redux/store'
+import { actionCreators } from './redux/auth/actions'
+
+/* Pages Components Imports */
+import IndexPage from './routes/IndexPage'
+import LoginPage from './routes/LoginPage'
+import LogoutPage from './routes/LogoutPage'
+import BoardPage from './routes/BoardPage'
+import PageNotFound from './routes/PageNotFound'
+
+/* Authenticate user */
+if (AUTH.isUserAuthenticated()) {
+    store.dispatch(
+        actionCreators.loginSuccess(AUTH.getUserToken()!)
```

Comment:
What is the "!" for ?

Reply:
getUserToken can return null, the `!` forces the evalutation of the variable like in Swift.